cmake/FindWayland.cmake

```diff
@@ -0,0 +1,14 @@
+find_package(PkgConfig)
+
+if(PKG_CONFIG_FOUND)
+    pkg_check_modules(WAYLAND_CLIENT REQUIRED IMPORTED_TARGET wayland-client)
+
+    if(WAYLAND_CLIENT_FOUND)
+        find_path(WAYLAND_CLIENT_INCLUDE_DIR NAMES wayland-client.h HINTS ${WAYLAND_CLIENT_INCLUDE_DIRS})
+        add_library(Wayland::client ALIAS PkgConfig::WAYLAND_CLIENT)
+    endif()
+
+    set(WAYLAND_INCLUDE_DIR ${WAYLAND_CLIENT_INCLUDE_DIR})
+    set(WAYLAND_LIBRARIES ${WAYLAND_CLIENT_LIBRARIES})
```

Comment:
You could add `WAYLAND_CLIENT_INCLUDE_DIR` to the list of `INTERFACE` include directories of the `Wayland::client` target, so that you don't then need to call both `target_link_libraries()` and `include_directories()` in code using the module

Reply:
Would that still be needed with the current changes?
Also, I think the include directory for `wayland-client` is already set after pkg_check_modules finds the module.